File: src/test/python/0003-mvm-video-1.py

class JiiifyPresentation0003_1:

  def __init__(self):
    # We pass in the version of the uber package to use when running this test
    import sys; sys.path.append('../../../target/jiiify-presentation-v3-' + sys.argv[0] + '-uber.jar')

    from info.freelibrary.iiif.presentation.v3 import Canvas, VideoContent, Manifest
    from info.freelibrary.iiif.presentation.v3.ids import MinterFactory
    from info.freelibrary.iiif.presentation.v3.properties import Label

    manifestID = 'https://iiif.io/api/cookbook/recipe/0003-mvm-video/manifest'
    videoID = 'https://fixtures.iiif.io/video/indiana/lunchroom_manners/high/lunchroom_manners_1024kb.mp4'

    manifest = Manifest(manifestID, Label('en', 'Video Example 3'))
    minter = MinterFactory.getMinter(manifest)
    canvas = Canvas(minter).setDuration(572.034).setWidthHeight(640, 360)
    videoContent = VideoContent(videoID).setDuration(572.034).setWidthHeight(480, 360)

    canvas.paintWith(minter, [videoContent])
    manifest.setCanvases([canvas])

    print(manifest)

JiiifyPresentation0003_1()

File: src/test/ruby/0003-mvm-video-1.rb

module JiiifyPresentation0003_1
  require 'jruby'

  # We pass in the version of the uber package to use when running this test
  require_relative '../../../target/jiiify-presentation-v3-' + ARGV[0] + '-uber.jar'

  include_package 'info.freelibrary.iiif.presentation.v3'
  include_package 'info.freelibrary.iiif.presentation.v3.ids'
  include_package 'info.freelibrary.iiif.presentation.v3.properties'

  manifestID = 'https://iiif.io/api/cookbook/recipe/0003-mvm-video/manifest'
  videoID = 'https://fixtures.iiif.io/video/indiana/lunchroom_manners/high/lunchroom_manners_1024kb.mp4'

  manifest = Manifest.new(manifestID, Label.new('en', 'Video Example 3'))
  minter = MinterFactory.getMinter(manifest)
  canvas = Canvas.new(minter).setWidthHeight(640, 360).setDuration(572.034)
  videoContent = VideoContent.new(videoID).setWidthHeight(480, 360)

  videoContent.setDuration(572.034)
  canvas.paintWith(minter, videoContent)
  manifest.setCanvases(canvas)

  puts manifest.toString()
end
